exasol/error/_cli.py

```diff
@@ -0,0 +1,354 @@
+import argparse
+import ast
+import json
+import sys
+from dataclasses import asdict, dataclass, is_dataclass
+from enum import IntEnum
+from itertools import chain
+from pathlib import Path
+from typing import List, Optional
+
+
+class ExitCode(IntEnum):
+    SUCCESS = 0
+    FAILURE = -1
+
+
+@dataclass(frozen=True)
+class Placeholder:
+    """
+    Placeholder according to schema specification.
+    https://schemas.exasol.com/error_code_report-1.0.0.json
+    """
+
+    placeholder: str
+    description: Optional[str]
+
+
+@dataclass
+class ErrorCodeDetails:
+    """
+    Error code details according to schema specification.
+    https://schemas.exasol.com/error_code_report-1.0.0.json
```

Comment:
To be honest I created a pydantic class and made a `dataclass` out of it. Once I converted it, I got  rid of the `pydantic` dependency, because I such a basic library should come with as less dependencies as possible.

Reply:
Agree